### PR TITLE
Improve `rebaseBranch` early-out case

### PR DESCRIPTION
**Pull request by noencke**
## Description

This makes `rebaseBranch` early-out (without doing any rebasing) in more cases. See the code comment for details.

### CODE REVIEW

File: experimental/dds/tree2/src/core/rebase/utils.ts [hunk 1]
```diff
@@ -162,12 +162,25 @@ export function rebaseBranch<TChange>(
 	// Figure out how much of the trunk to start rebasing over.
 	const targetRebasePath = targetPath.slice(0, newBaseIndex + 1);
 	const newSourceCommits: GraphCommit<TChange>[] = [...targetRebasePath];
+	/** The commit on the target branch that the new source branch branches off of */
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/core/rebase/utils.ts:165]: Is this the same as the closest common ancestor?

Comment (noencke) [reply experimental/dds/tree2/src/core/rebase/utils.ts:165]: Yes

File: experimental/dds/tree2/src/core/rebase/utils.ts [hunk 2]
```diff
@@ -162,12 +162,25 @@ export function rebaseBranch<TChange>(
...
 	const newBase = targetPath[newBaseIndex];
-	// Early out if the source branch's commits and the commits it is being rebased over are equivalent,
-	// i.e. they have exactly the same revision tags in the same order.
-	if (sourceSet.size === 0 && sourcePath.length === targetRebasePath.length) {
+
+	// If all commits that are about to be rebased over on the target branch already comprise the start of the source branch,
+	// are in the same order, and have no other commits interleaving them, then no rebasing needs to occur. Those commits can
+	// simply be removed from the source branch, and the remaining commits on the source branch are reparented off of the new
+	// base commit
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/core/rebase/utils.ts:171]:
```suggestion
	// base commit.
```

File: experimental/dds/tree2/src/core/rebase/utils.ts (hunk 1 above)
Comment (noencke) [reply experimental/dds/tree2/src/core/rebase/utils.ts]: I added a clarifying statement

**Review by alex-pardes — APPROVED** [review]

**Pull request merged**